File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/GremlinThief_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class GremlinThief_Model extends MonsterInfo{
    public static Id = "GremlinThief"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(11, 15);
        else
            return Mathf.RandomRange(10, 14);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/thiefGremlin/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/thiefGremlin/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/thiefGremlin/skeleton.png"));
    }

    get Name(): string {
        return "卑鄙地精";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/GremlinThief_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class GremlinThief_Model : MonsterInfo
    {
        public static string Id = "GremlinThief";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(11, 15) : Random.Range(10,14);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/thiefGremlin/skeleton.atlas",
                    "Monsters/theBottom/thiefGremlin/skeleton.json", "Monsters/theBottom/thiefGremlin/skeleton.png");
            }
        }

        public override string Name => "卑鄙地精";

        public GremlinThief_Model(Vector2 pos) : base(pos)
        {
        }
    }
}